compiler/AST/symbol.cpp

```diff
@@ -606,7 +549,9 @@ GenRet VarSymbol::codegen() {
   ret.chplType = typeInfo();
 
   if( outfile ) {
-    if (immediate) {
+    // dtString immediates don't actually codegen as immediates, we just use
+    // them for param string functionality.
```

Comment:
This something that has changed a decent bit since this PR was opened. See #2326 for the details. The gist of it is that dtString literals are locale private, so it should be faster than just creating one in the body of a function when necessary. The code is a bit simpler now too.